src/index.spec.ts

```diff
@@ -0,0 +1,301 @@
+import fetchMock from "jest-fetch-mock";
+import { mockRandomForEach } from "jest-mock-random";
+
+import { Fetch, TestResultBundle } from "@openinsights/openinsights";
+import { Provider } from "./";
+import * as template from "./lib/templateResource";
+
+import configFixture from "./fixtures/config";
+import clientInfoFixture from "./fixtures/clientInfo";
+import entryFixture from "./fixtures/entry";
+
+const settingsFixture = configFixture.settings;
+const taskFixture = configFixture.tasks[1];
+
+describe("FastlyProvider", (): void => {
+  let providerFixture: Provider;
+
+  beforeEach((): void => {
+    providerFixture = new Provider(settingsFixture) as Provider;
+    providerFixture.setSessionConfig(configFixture);
+    fetchMock.resetMocks();
+  });
+
+  describe("shouldRun", (): void => {
+    it("should always run", () => {
+      const result = new Provider(settingsFixture);
+      expect(result.shouldRun()).toBeTruthy();
+    });
+  });
+
+  describe("fetchSessionConfig", (): void => {
+    it("should fetch session config", (): Promise<void> => {
+      fetchMock.mockResponseOnce(JSON.stringify(configFixture), {
+        headers: {
+          "access-control-allow-origin": "*",
+        },
+      });
+      const provider = new Provider(settingsFixture);
+      return provider.fetchSessionConfig().then((result) => {
+        expect(result).toEqual(configFixture);
+      });
+    });
+
+    it("should retry the fetch if it fails", (): Promise<void> => {
+      fetchMock.mockAbortOnce();
+      fetchMock.mockResponseOnce(JSON.stringify(configFixture), {
+        headers: {
+          "access-control-allow-origin": "*",
+        },
+      });
+      const provider = new Provider(settingsFixture);
+      return provider.fetchSessionConfig().then((result) => {
+        expect(fetchMock.mock.calls).toHaveLength(2);
+        expect(result).toEqual(configFixture);
+      });
+    });
+  });
+
+  describe("expandTasks", (): void => {
+    mockRandomForEach([0.5]);
+    it("should expand session config to tasks", (): void => {
+      const result = providerFixture.expandTasks();
+      expect(result).toHaveLength(3);
```

Comment:
The magic number `3` is not very clear here. Can we just use `config.tasks.length` ?